.github/workflows/test-publish-dry-run.yaml

```diff
@@ -19,21 +19,21 @@ jobs:
         gradle: ["7.2"]
         java: ["11"]
         include:
-          - agp: "4.1.3"
-            gradle: "6.8.3"
-            java: "8"
-          - agp: "4.1.3"
-            gradle: "6.8.3"
-            java: "8"
-          - agp: "4.1.3"
-            gradle: "7.0.2"
-            java: "8"
-          - agp: "4.2.2"
-            gradle: "6.8.3"
-            java: "8"
-          - agp: "4.2.2"
-            gradle: "7.0.2"
-            java: "8"
+#          - agp: "4.1.3"
+#            gradle: "6.8.3"
+#            java: "8"
+#          - agp: "4.1.3"
+#            gradle: "6.8.3"
+#            java: "8"
+#          - agp: "4.1.3"
+#            gradle: "7.0.2"
+#            java: "8"
+#          - agp: "4.2.2"
+#            gradle: "6.8.3"
+#            java: "8"
+#          - agp: "4.2.2"
+#            gradle: "7.0.2"
+#            java: "8"
```

Comment:
agp 4.x has to be dropped due to AndroidComponentsExtension

Reply:
mm, in theory this class still exists, but was deprecated, if we want to maintain compatibility, we could use both depending on the AGP version - https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-api/src/main/java/com/android/build/api/extension/AndroidComponentsExtension.kt;bpv=1